### PR TITLE
SQL sync first cut

### PR DESCRIPTION
**Pull request by oscarbatori**
WIP still, but tests suite is green.

To do:
- [ ] implement for Postgres
- [ ] get Dolt diff based features working
- [ ] sort out Doltpy management of connections
- [ ] RDBMS to Dolt sync
- [ ] docs

I reckon these will be relatively straight forward to get something basic working since the test harness is now setup.

### CODE REVIEW

File: doltpy/etl/sql_sync/dolt.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    return inner
+
+
+def get_target_writer() -> TargetWriter:
+    raise NotImplemented()
+
+
+def read_from_table(table_name: str, repo: Dolt, latest: bool = True, commit_ref: str = None) -> List[tuple]:
+    """
+    This function pulls data from Dolt in three different modes:
```
Comment (zachmu) [inline doltpy/etl/sql_sync/dolt.py:44]: This smells funny. You have 2 bits of state and only 3 possible modes of operation. Yes, one mode throws an error, but still.

I would make this two functions: one for reading the whole table, and another for reading a diff, where ref = None means "give me the most recent diff"

File: doltpy/etl/sql_sync/tests/fixtures/dolt.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    curs1 = connection.cursor()
+    curs1.execute(CREATE_TEST_TABLE)
+    connection.commit()
+
+    yield repo, TABLE_NAME
+
+    # The SQL server seems to get into a strange state
+    repo.stop_server()
+    repo.start_server()
+    new_conn = repo.cnx
```
Comment (zachmu) [inline doltpy/etl/sql_sync/tests/fixtures/dolt.py:27]: Shouldn't this be connection = repo.start_server()?

File: doltpy/etl/sql_sync/dolt.py (hunk 1 above)
Comment (oscarbatori) [reply doltpy/etl/sql_sync/dolt.py:44]: @zachmu good feedback, thanks

**Comment by oscarbatori** [comment]
I ended up having to make a fair number of updates to this, including:
- implementing updates and deletes for when Dolt repo has had update or delete operations applied
- removing Postgres support as I need to do a bit more work on getting updates and deletes working
- fix some miscellaneous issues in Doltpy, most importantly the feature of sitting on connection objects, instead clients should request a connection object

**Pull request merged**